mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -225,8 +225,22 @@ impl StreamWakeableState for RingBuffer {
         let result = load_block_header(&mut self.file, start, block_size, self.max_file_size);
         if let Ok(block_header) = result {
             let BlockVersion::Version1(inner) = block_header.inner();
+            // It is possible that we end up writing a smaller block on wrap-around,
+            // where the file pointers will end up being in the middle of random data.
+            // To avoid this, we need to check if we have a block of data, if we do,
+            // we will need to just allow the write. Above cases will prevent us from
+            // accidentally corrupting.
+            let mut allow_write = false;
```

Comment:
In general, mutating is error-prone esp. in multithreaded runtime. I usually try to prioritize immutable over mutable operations whenever possible.

```rust
let allow_write = bin_block.is_zeroed() && inner.hint() != BLOCK_HINT;
```

Reply:
As what Vadim mentioned above, I think this section can be dropped entirely.